river/sync.go

```diff
@@ -333,9 +333,22 @@ func (r *River) makeReqColumnData(col *schema.TableColumn, value interface{}) in
 	case schema.TYPE_DATETIME, schema.TYPE_TIMESTAMP:
 		switch v := value.(type) {
 		case string:
-			vt, _ := time.ParseInLocation(mysql.TimeFormat, string(v), time.Local)
+			vt, err := time.ParseInLocation(mysql.TimeFormat, string(v), time.Local)
+			if err != nil || vt.IsZero() { // failed to parse date or zero date
+				return nil
+			}
 			return vt.Format(time.RFC3339)
 		}
+	case schema.TYPE_DATE:
+		switch v := value.(type) {
+		case string:
+			dateFormat := "2006-01-02"
+			vt, err := time.Parse(dateFormat, string(v))
+			if err != nil || vt.IsZero() { // failed to parse date or zero date
```

Comment:
if we meet 0000-00-00, is it ok to insert nil to ES?

can you help me add some tests?

Reply:
zero times or invalid date should return nil, which will send null to ES.

I have added some tests